models/Review.js

```diff
@@ -44,6 +44,7 @@ ReviewSchema.statics.calculateAverageRating = async function (productId) {
     { $match: { product: productId } }, //1stage Match reviews with the specified productId
     {
       $group: { //2 stage- Groups the matched reviews together - this is what we return
+        // AKOSREVIEW: reword/explaining group id
         _id: null, // this is id of the group- can be product id, rating, etc //we use null or _id: '$product' // but  if we want to calculate how many reviews we have with specific rating for this product: _id : '$rating'  //check syntax  + add: ,amount:{$sum:1}- see code below line 81
         averageRating: { $avg: '$rating' }, // Calculate average rating  using the $avg aggregation operator. 
```

Comment:
When using the `$group` aggregator, the `_id` field tells Mongo what to group the results by. If you pass in `null` or a constant value (like `1`), it will just group _all_ the results together. In this case, we've already filtered the results in the first aggregate step to give us just the reviews for a particualr product; we're then grouping together all of those reviews, and doing some calculations / "aggregations" on those groups (to get the average rating and number of reviews).